testing/src/main/java/io/stargate/it/grpc/streaming/ExecuteBatchStreamingTest.java

```diff
@@ -116,31 +130,45 @@ public void onCompleted() {}
                     rowOf(Values.of("c"), Values.of(3)))));
   }
 
+  // From [https://github.com/stargate/stargate/pull/1647] to reproduce
+  // issue [https://issues.apache.org/jira/browse/CASSANDRA-17401]
+
+  @Test
+  public void repeatedStreamingBatch(@TestKeyspace CqlIdentifier keyspace) {
+    List<StreamingResponse> responses = new CopyOnWriteArrayList<>();
+
+    StargateStub stub = asyncStubWithCallCredentials();
+    StreamObserver<Batch> requestObserver =
+        stub.executeBatchStream(new StreamObserverWithList(responses));
+    final int ROUNDS = 1000;
+
+    for (int i = 0; i < ROUNDS; i++) {
+      requestObserver.onNext(
+          Batch.newBuilder()
+              .addQueries(cqlBatchQuery("INSERT INTO test (k, v) VALUES ('a', 1)"))
+              .addQueries(
+                  cqlBatchQuery(
+                      "INSERT INTO test (k, v) VALUES (?, ?)", Values.of("b"), Values.of(2)))
+              .setParameters(batchParameters(keyspace))
+              .build());
+    }
+
+    Awaitility.await()
+        .atMost(30, TimeUnit.SECONDS)
+        .untilAsserted(() -> assertThat(responses.size()).isEqualTo(2 * ROUNDS));
```

Comment:
Fails here only getting 1000 -- but I don't think that actually reproduces the NPE we want to catch; both since no matching error found in logs and since suggested fix for `QueryProcessor` access does not solve the fail.
So I suspect assertion here is incorrect.